pkg/gofr/http/middleware/metrics.go

```diff
@@ -21,11 +22,19 @@ type metrics interface {
 func Metrics(metrics metrics) func(inner http.Handler) http.Handler {
 	return func(inner http.Handler) http.Handler {
 		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			// Compile regex for common file extensions
+			staticFileRegex := regexp.MustCompile(`\.(css|js|png|jpg|jpeg|gif|ico|svg|txt|html|json|woff|woff2|ttf|eot|pdf)$`)
+
 			start := time.Now()
 
 			srw := &StatusResponseWriter{ResponseWriter: w}
 
 			path, _ := mux.CurrentRoute(r).GetPathTemplate()
+
+			if path == "/" || strings.HasPrefix(path, "/static") || staticFileRegex.MatchString(r.URL.Path) {
+				path = r.URL.Path
+			}
+
```

Comment:
Please don't use regexp for this.

You can use [filepath.Ext](https://pkg.go.dev/path/filepath#Ext)

Something like this could work

```suggestion
			switch strings.ToLower(filepath.Ext(r.URL.Path)) {
				case "css", "js", "png", "jpg","jpeg","gif","ico","svg","txt","html","json","woff","woff2","ttf","eot","pdf":
					path = r.URL.Path
			}
			if path == "/" || strings.HasPrefix(path, "/static") {
				path = r.URL.Path
			}
```

Reply:
Sure i have used `filepath.Ext()` method to match it instead of regex as per your suggestion.